### DISCUSSION TITLE
app.exception_handler does not work when used in separate file

### DISCUSSION THREAD

**Discussion by arunbhati** [aws-powertools/powertools-lambda-python#1191] [Q&A] [answered]
Hi, 

I am using **APIGatewayRestResolver** and I'm having trouble working with **@app.exception_handler**. When I have **APIGatewayRestResolver** object and **@app.exception_handler** decorator in same file then it works but if I create a separate file for exceptions(**exceptions.py**) and do the import of APIGatewayRestResolver then it doesn't work. 
Following are steps to reproduce.
1. Create a lambda function in Python with Lambda Layer for AWS Powertools from [here](https://awslabs.github.io/aws-lambda-powertools-python/latest/).
2. Attach this Zip file to lambda function - Zip file: [Archive.zip](https://github.com/awslabs/aws-lambda-powertools-python/files/8587999/Archive.zip)
3. Use following input for Lambda request
```
{
  "resource": "/proxy",
  "path": "/proxy",
  "httpMethod": "GET",
  "headers": null
}
```
4. Lambda function should fail with following error
```
{
  "errorMessage": "",
  "errorType": "ValueError",
  "requestId": "24f06853-16fc-4657-9360-40409f0726be",
  "stackTrace": [
    "  File \"/var/task/lambda_function.py\", line 10, in lambda_handler\n    return axis_controllers.main(event, context)\n",
    "  File \"/var/task/src/abc/axis_controllers.py\", line 11, in main\n    return app.resolve(event, context)\n",
    "  File \"/opt/python/aws_lambda_powertools/event_handler/api_gateway.py\", line 498, in resolve\n    return self._resolve().build(self.current_event, self._cors)\n",
```
5. Now move **ValueError** exception method from **exceptions.py** into **__init__.py** file in same folder.
6. Rerun test with same input and now lambda output will show expected result(shown below).
7.
```
{
  "statusCode": 400,
  "headers": {
    "Content-Type": "text/plain"
  },
  "body": "Invalid request",
  "isBase64Encoded": false
}
```

I don't have much expertise on how decorator works so any help here would be appreciated.
Thanks.

**Comment by michaelbrewer** [comment, marked as answer]
@arunbhati in your example nothing is referencing `exceptions.py` so it would not call either of the `@app.exception_handler` or `@app.not_found` decorators.

You can verify that `exceptions.py` is not being referenced by adding a `print` statement. 

So maybe either reference the `exceptions.py` as an import elsewhere or load it via `__init__.py`, or move the logic in `__init__.py__`
